cmd/keymasterd/certgen.go

```diff
@@ -217,10 +228,17 @@ func (state *RuntimeState) postAuthSSHCertHandler(
 }
 
 func (state *RuntimeState) getUserGroups(username string) ([]string, error) {
+	if state.gitDB != nil {
```

Comment:
I dont like the locaiton of this I would probably split this function into 3: getUserGrouos, getLdapUserGroups and getGitDBUserGroups. And only call getGitDBUserGRoups if ldap config is not present. I also think we should add an "experimental flag" so that these kind of changes can get rolled in with less friction. At this moment I dont like the gitDB, but I cannot pinpoint why.

Reply:
OK, I've split them up and gave LDAP higher configuration priority.

I don't want to add an extra flag to enable the GitDB feature because it makes deployment more painful. Since this is gated by a configuration entry, an extra flag would be superflous.